letsencrypt/plugins/common.py

```diff
@@ -104,14 +104,24 @@ class Addr(object):
     :param str port: port number or \*, or ""
 
     """
-    def __init__(self, tup):
+    def __init__(self, tup, ipv6=False):
         self.tup = tup
+        self.ipv6 = ipv6
 
     @classmethod
     def fromstring(cls, str_addr):
         """Initialize Addr from string."""
-        tup = str_addr.partition(':')
-        return cls((tup[0], tup[2]))
+        if str_addr.startswith('['):
+            # ipv6 addresses starts with [
+            endIndex = str_addr.rfind(']')
+            host = str_addr[:endIndex + 1]
+            port = ''
+            if len(str_addr) > endIndex + 2 and str_addr[endIndex + 1] == ':':
```

Comment:
No it's actually not. I'd suspect ports < 10 are failing currently. Going to fix that, good catch.